### ISSUE TITLE
Page not found for post_tag sitemap

### ISSUE THREAD

**Issue by Koli14** [GoogleChromeLabs/wp-sitemaps#167]
**Describe the bug**
If I visit /wp-sitemap.xml, there is an url in the list: /wp-sitemap-taxonomies-post_tag-1.xml
If I click on it, 404 page comes in. All other links are working (like: /wp-sitemap-taxonomies-category-1.xml)

**To Reproduce**
1. Go to https://wp.szinhaz.online/wp-sitemap.xml
2. Click on https://wp.szinhaz.online/wp-sitemap-taxonomies-post_tag-1.xml
4. See my blank 404 page

**Expected behavior**
I expect it to work like for other taxonomies

**Desktop (please complete the following information):**
 - OS: Linux 4.4.0-177-generic x86_64 Ubuntu 18.04.3 LTS

**Additional context**
My site is a headless WP, but I don't think it's matters.

**Comment by swissspidy** [comment]
@Koli14 Thanks for your report. Does this still happen with the current master branch?

We might need to check the rewrite rules regarding underscores...

**Comment by swissspidy** [comment]
If it's because of the underscore, might be a duplicate of #166.

cc @pbiron

**Comment by pbiron** [comment]
I can't replicate with v0.2.0, but can with master.

The regex for the rewrite rule in 0.2.0 is `^wp-sitemap-taxonomies-([A-z]+)-?([0-9]+)?\.xml$'`, which includes `_`, but kind of by "accident", because it's in the range of characters between the uppercase  and lowercase letters (so is `[`, `^` and a few others that probably weren't intended).

The new regex in master (after #150) is `^wp-sitemap-([a-z]+?)-([a-z\d-]+?)-(\d+?)\.xml$`, which is clear we forgot to include `_`.

So, I don't think it's a duplicate of #166, but it is related in that we need to make sure that we need to make sure the regex matches all-and-only the sitemaps that can be registered.

I'll do a quick PR to add `_` to the sub-type regex clause, which will fix things for the built-in `post_tag` taxonomy, but we'll need to to address the problem more generally in a solution to #166.
